plugins/search-backend-node/src/engines/LunrSearchEngine.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  SearchQuery,
+  IndexableDocument,
+  SearchResultSet,
+} from '@backstage/search-common';
+import lunr from 'lunr';
+import { Logger } from 'winston';
+import { SearchEngine, QueryTranslator } from '../types';
+
+type ConcreteLunrQuery = {
+  lunrQueryString: string;
+  documentTypes: string[];
+};
+
+export class LunrSearchEngine implements SearchEngine {
+  protected lunrIndices: Record<string, lunr.Index> = {};
+  protected docStore: Record<string, IndexableDocument>;
+  protected logger: Logger;
+
+  constructor({ logger }: { logger: Logger }) {
+    this.logger = logger;
+    this.docStore = {};
+  }
+
+  translator: QueryTranslator = ({
+    term,
+    filters,
+  }: SearchQuery): ConcreteLunrQuery => {
+    let lunrQueryFilters;
+    if (filters) {
+      lunrQueryFilters = Object.keys(filters)
+        .map(key => ` +${key}:${filters[key]}`)
+        .join('');
```

Comment:
```suggestion
      lunrQueryFilters = Object.entries(filters)
        .map(([key, value]) => ` +${key}:${value}`)
        .join('');
```

entries can be a little neater